android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java

```diff
@@ -290,41 +290,60 @@ public void handleAppLink(String uri, Promise promise) {
     @ReactMethod
     public void trackInAppOpen(String messageId, @Nullable Integer location) {
         IterableInAppMessage message = RNIterableInternal.getMessageById(messageId);
+
         if (message == null) {
-            IterableLogger.d(TAG, "Failed to get InApp for message id : " + messageId);
+            IterableLogger.d(TAG, "Failed to get in-app for message ID: " + messageId);
             return;
         }
+
         IterableApi.getInstance().trackInAppOpen(message, Serialization.getIterableInAppLocationFromInteger(location));
     }
 
     @ReactMethod
     public void trackInAppClick(String messageId, @Nullable Integer location, String clickedUrl) {
         IterableInAppMessage message = RNIterableInternal.getMessageById(messageId);
         IterableInAppLocation inAppOpenLocation = Serialization.getIterableInAppLocationFromInteger(location);
+
         if (message == null) {
-            IterableLogger.d(TAG, "Failed to get InApp for message id : " + messageId);
+            IterableLogger.d(TAG, "Failed to get in-app for message ID: " + messageId);
             return;
         }
+
         if (clickedUrl == null) {
-            IterableLogger.d(TAG, "clickedURL is null");
+            IterableLogger.d(TAG, "clickedUrl is null");
```

Comment:
Here we are return and stop the process if clickedUrl is null. Should we not pass and let it stay null?

Reply:
That's specific to `trackInAppClick`, which needs it for the event (otherwise the event doesn't make sense)